examples/injection/proto/injection.proto

```diff
@@ -0,0 +1,69 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+import "oak/proto/handle.proto";
+
+package oak.examples.injection;
+
+message BlobStoreInterface {
+  oak.handle.Sender sender = 1 [(oak.handle.message_type) = ".oak.examples.injection.BlobRequest"];
+  oak.handle.Receiver receiver = 2
+      [(oak.handle.message_type) = ".oak.examples.injection.BlobResponse"];
+}
+
+message GetBlobRequest {
+  fixed64 id = 1;
+}
+
+message PutBlobRequest {
+  // Put 0 for new blob to store.
```

Comment:
So the reason that it works like this is that it allows both inserts and updates through the same `PutBlobRequest`. For new inserts a client can leave this field unset (i.e. default '0' value). For updates the client can specify the id of the blob to modify.

If we allow clients to specify an id they like, then we have to complicate things (slightly) and store blobs in a map instead of a vec.

What exactly do you mean by shifting indices up and down?